src/main/java/org/example/chess/piece/Horse.java

```diff
@@ -0,0 +1,64 @@
+package org.example.chess.piece;
+
+import org.example.chess.Field;
+import org.example.chess.piece.util.ChessPieceFieldValidator;
+
+import java.util.HashSet;
+import java.util.Set;
+
+final class Horse extends ChessPiece{
+    Horse(Color color, Field currentPosition) {
+        super(color, currentPosition);
+    }
+
+    @Override
+    Set<Field> possibleMoves() {
+        Set<Field> fieldsOfPossibleMoves = new HashSet<>();
+
+        checkMoves(fieldsOfPossibleMoves, "up");
+        checkMoves(fieldsOfPossibleMoves, "down");
+        checkMoves(fieldsOfPossibleMoves, "right");
+        checkMoves(fieldsOfPossibleMoves, "left");
+
+        return fieldsOfPossibleMoves;
+    }
+
+
+    private static Set<Field> checkMoves(Set<Field> fields, String command) {
+        int cycles = 2;
+        int row = 5;
+        int col = 3;
```

Comment:
I guess  row and col values shouldn't be hard-coded but rather obtain from currentPosition field